From f9488c866b0f4b60856a6d51077997a521f55c1f Mon Sep 17 00:00:00 2001
From: Michael van der Westhuizen <r1mikey@gmail.com>
Date: Tue, 8 Sep 2015 11:34:22 +0200
Subject: [PATCH 4/5] The picoxcell read/write register functions are
 completely unnecessary, so define them to __raw_readl and __raw_writel, since
 that's what they do

---
 arch/arm/cpu/arm926ejs/picoxcell/utilities.c    | 12 ------------
 arch/arm/include/asm/arch-picoxcell/utilities.h |  6 +++---
 2 files changed, 3 insertions(+), 15 deletions(-)

diff --git a/arch/arm/cpu/arm926ejs/picoxcell/utilities.c b/arch/arm/cpu/arm926ejs/picoxcell/utilities.c
index 2e5cb23..23f79e1 100644
--- a/arch/arm/cpu/arm926ejs/picoxcell/utilities.c
+++ b/arch/arm/cpu/arm926ejs/picoxcell/utilities.c
@@ -58,18 +58,6 @@ static int is_memif_arm_present_on_pc30xx (void);
 static int is_memif_arm_disabled (void);
 
 /* Functions --------------------------------------------------------------- */
-__inline unsigned int picoxcell_read_register (const unsigned int address)
-{
-	/* Read an integer (32 bit) value from a register */
-	return (*(volatile unsigned int *)address);
-}
-
-__inline void
-picoxcell_write_register (const unsigned int value, const unsigned int address)
-{
-	/* Write an integer (32 bit) value to a register */
-	*(volatile unsigned int *)address = value;
-}
 
 const char *picoxcell_get_partname (void)
 {
diff --git a/arch/arm/include/asm/arch-picoxcell/utilities.h b/arch/arm/include/asm/arch-picoxcell/utilities.h
index 6f55277..b0c3643 100644
--- a/arch/arm/include/asm/arch-picoxcell/utilities.h
+++ b/arch/arm/include/asm/arch-picoxcell/utilities.h
@@ -19,6 +19,7 @@
 
 /* Includes ---------------------------------------------------------------- */
 #include <common.h>
+#include <asm/io.h>
 
 /* Prototypes--------------------------------------------------------------- */
 /*!
@@ -27,7 +28,7 @@
  * \param The address to read from
  * \return The value read
  */
-__inline unsigned int picoxcell_read_register (const unsigned int address);
+#define picoxcell_read_register(a) __raw_readl((a))
 
 /*!
  * \brief Write a 32 bit value to a register
@@ -35,8 +36,7 @@ __inline unsigned int picoxcell_read_register (const unsigned int address);
  * \param The value to write.
  * \param The address to write to
  */
-__inline void
-picoxcell_write_register (const unsigned int value, const unsigned int address);
+#define picoxcell_write_register(v,a) __raw_writel((v),(a))
 
 /*!
  * \brief Read the device identification code from the axi2cfg block
-- 
2.1.4

